src/test/bindc/bindc.yaml

```diff
@@ -7,5 +7,5 @@ hosts:
   testnode:
     network_node_id: 0
     processes:
-    - path: test-bind
+    - path: ./test-bind
```

Comment:
Ah drat; I didn't realize we currently interpret `x` as `./x`. That makes this change a breaking change, since current yaml files will now have different behavior - either failing to run at all if the binary isn't in the `PATH`, or worse potentially running a different binary than they ran before.

Technically we could make this a non-breaking change by still searching the current directory first, but I don't think that's the behavior we ultimately want.

I think the best way to handle this for now is to put this feature behind an [`experimental`](https://shadow.github.io/docs/guide/shadow_config_spec.html#experimental) flag like `--enable-searching-path`  that's disabled by default. In the next major release we can change the default and/or remove the flag.

@stevenengler @robgjansen thoughts?

Reply:
We could also keep everything the way it is and use an additional field like `exec: ` in the yaml which will (if `path: ` is not specified) search for its value in `$PATH` and set the path field in the processed-config.yaml.

Reply:
Interesting idea, but I think ultimately the distinction between `path` and `exec` wouldn't be obvious based on the names.

The [`exec` man-page](https://www.man7.org/linux/man-pages/man3/exec.3.html) uses `file` vs `path`... but calls the old behavior we want to preserve `file`. So I don't think adding a `file` field and having that do the `$PATH` search would make sense.

I still lean towards enabling `$PATH`-search via a feature-flag

Reply:
Actually another option is:

* Always search the current directory first for now (effectively prepend the current working directory to the path)
* Log a warning if the executable was found there via path-search, and wouldn't have been otherwise. (i.e. didn't have the `./` prefix and `PATH` didn't include the current directory) 
* In 3.0 stop searching the current directory first

That's maybe a slightly nicer migration strategy, though the logic for when to log a warning is a little tricky

Reply:
Hmm, I kinda like that idea to search in CWD before resolving something using PATH as a temporary measure. I should be able to do that.

Reply:
OK; I haven't heard from the others yet, but think it's probably safe to go forward with that plan.

I think the warning-logic isn't that bad after thinking about it some more. we warn if all of:

* There's no `./` prefix
* We found it in the CWD
* PATH-expansion *without* the CWD prefix gave a different result - either didn't find the binary at all or found a different one.

Reply:
Either with an experimental flag or prepending "." to the path search string (and removing in 3.0) sounds fine to me. But another case to handle is non-absolute paths that contain a separator. For example `- path: foo/test-bind` should also find the `test-bind` binary in directory `foo`.

Reply:
Both the experimental flag and the CWD prepend are good options. FWIW, I also lean toward the prepend CWD plan, and it would be great if the code is written such that it is really simple to remove that behavior when we reach 3.0 (or with sufficient comments so we understand/remember what needs to be changed later).